ludwig/trainers/trainer_lightgbm.py

```diff
@@ -319,51 +308,149 @@ def run_evaluation(
         progress_tracker: ProgressTracker,
+        progress_bar: LudwigProgressBar,
         save_path: str,
-    ) -> lgb.Booster:
-        name_to_metrics_log = {
-            LightGBMTrainer.TRAIN_KEY: progress_tracker.train_metrics,
-            LightGBMTrainer.VALID_KEY: progress_tracker.validation_metrics,
-            LightGBMTrainer.TEST_KEY: progress_tracker.test_metrics,
-        }
-        tables = OrderedDict()
+        training_set: Union["Dataset", "RayDataset"],  # noqa: F821
+        validation_set: Union["Dataset", "RayDataset"],  # noqa: F821
+        test_set: Union["Dataset", "RayDataset"],  # noqa: F821
+        train_summary_writer: SummaryWriter,
+        validation_summary_writer: SummaryWriter,
+        test_summary_writer: SummaryWriter,
+    ) -> bool:
+        self.callback(lambda c: c.on_batch_start(self, progress_tracker, save_path))
+
+        booster = None
+        evals_result = {}
+        booster = self.train_step(
+            params, lgb_train, eval_sets, eval_names, booster, self.boosting_rounds_per_checkpoint, evals_result
+        )
+
+        progress_bar.update(self.boosting_rounds_per_checkpoint)
+        progress_tracker.steps += self.boosting_rounds_per_checkpoint
+        progress_tracker.last_improvement_steps = booster.best_iteration
+
+        # convert to pytorch for inference
+        self.model.lgb_booster = booster
+        self.model.compile()
+        self.model = self.model.to(self.device)
+
         output_features = self.model.output_features
         metrics_names = get_metric_names(output_features)
-        for output_feature_name, output_feature in output_features.items():
-            tables[output_feature_name] = [[output_feature_name] + metrics_names[output_feature_name]]
-        tables[COMBINED] = [[COMBINED, LOSS]]
-        booster = None
+        output_feature_name = next(iter(output_features))
+
+        loss_name = params["metric"][0]
+        loss = evals_result["train"][loss_name][-1]
+        loss = torch.tensor(loss, dtype=torch.float32)
+
+        should_break = self.run_evaluation(
+            training_set,
+            validation_set,
+            test_set,
+            progress_tracker,
+            train_summary_writer,
+            validation_summary_writer,
+            test_summary_writer,
+            output_features,
+            metrics_names,
+            save_path,
+            loss,
+            {output_feature_name: loss},
+        )
+
+        self.callback(lambda c: c.on_batch_end(self, progress_tracker, save_path))
 
-        for epoch, steps in enumerate(range(0, self.num_boost_round, self.boosting_round_log_frequency), start=1):
-            progress_tracker.epoch = epoch
+        return should_break
 
-            evals_result = {}
-            booster = self.train_step(
-                params, lgb_train, eval_sets, eval_names, booster, self.boosting_round_log_frequency, evals_result
-            )
+    def check_progress_on_validation(
```

Comment:
Is this function the same as the ECD trainer? If so, we should factor this out, like you've done with append_metrics.

Reply:
Not entirely, ECD trainer has additional logic here for reducing learning rate and increasing batch size